worker/src/lib/managers/RequestResponseManager.ts

```diff
@@ -0,0 +1,129 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { S3Client } from "../clients/S3Client";
+import { err, Result } from "../util/results";
+import { v4 as uuidv4 } from "uuid";
+import { Database } from "../../../supabase/database.types";
+
+export type RequestResponseContent = {
+  requestId: string;
+  organizationId: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  requestBody: any;
+  responseBody: string;
+};
+
+export class RequestResponseManager {
+  constructor(
+    private s3Client: S3Client,
+    private supabase: SupabaseClient<Database>
+  ) {}
+
+  async storeRequestResponseData({
+    organizationId,
+    requestId,
+    requestBody,
+    responseBody,
+  }: RequestResponseContent): Promise<Result<string, string>> {
+    const url = this.s3Client.getRequestResponseUrl(requestId, organizationId);
+
+    return await this.s3Client.store(
+      url,
+      JSON.stringify({ requestBody, responseBody })
+    );
+  }
+  async storeRequestResponseImage({
+    organizationId,
+    requestId,
+    requestBody,
+    responseBody,
+  }: RequestResponseContent): Promise<Result<string, string>> {
+    const uploadPromises = [];
+
+    for (const message of requestBody.messages) {
+      for (const item of message.content) {
+        if (item.type === "image_url") {
+          const imageUrl = item.image_url.url;
+          const assetId = uuidv4();
+
+          const uploadPromise = (async () => {
```

Comment:
Can we move this to a helpers file or another function? It is really hard to see what is passed into this scope